src/pages/signup.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react'
+import { useForm } from "react-hook-form";
+export default function SignUp() {
+    const { register, handleSubmit, formState: { errors } } = useForm();
+    const [data,setData]=useState({})
+    const [array,setArray]=useState([])
```

Comment:
use descriptive Names for variables.
ie. on Line 6 given name is ( array ) which is confusing so please use the name wisely